### PR TITLE
env variables for common configurations

### PR DESCRIPTION
**Pull request by khou**
Should allow the user to specify settings within stack YAML environments instead of having to modify the config.yml itself.

### CODE REVIEW

File: config/dashbase-tables/monitor/config.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dashbase:
     expirationHandler:
       params:
       type: purge
-    numDays: 7
+    numDays: ${RETENTION_NUM_DAYS:-7}
```
Comment (javasoze) [inline config/dashbase-tables/monitor/config.yml]: -7 as in negative 7?

Comment (khou) [reply config/dashbase-tables/monitor/config.yml]: http://www.dropwizard.io/0.9.3/docs/manual/core.html#environment-variables
env substitution with a default when not available!

File: config/dashbase-tables/quickstart/conf/config.yml [hunk 2]
```diff
@@ -19,12 +19,12 @@ dashbase:
...
   indexer:
-    numIndexingThreads: 3
+    numIndexingThreads: ${NUM_INDEXING_THREADS:-3}
     #maxNumConcurrentBuilding: 3
-    maxBufferDelayInSec: 90
+    maxBufferDelayInSec: ${MAX_BUFFER_DELAY_SEC:-90}
 
   retention:
-    numDays: .01
+    numDays: ${RETENTION_NUM_DAYS:-1}
```
Comment (keitaf) [inline config/dashbase-tables/quickstart/conf/config.yml]: so we cannot use 0.01 as default value?

**Comment by khou** [comment]
migrating to deployment repo

**Pull request closed without merging**